plugin/src/main/java/org/opensearch/ml/action/agents/GetAgentTransportAction.java

```diff
@@ -66,6 +75,9 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLAgen
                     try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                         MLAgent mlAgent = MLAgent.parse(parser);
+                        if (mlAgent.getIsHidden() && !isSuperAdmin) {
+                            mlAgent.removeModelId();
```

Comment:
If tool contains model then it will still be included as part of response. 

It is better to follow existing behavior of Get hidden model and not return complete agent information.

Reply:
For hidden model , we don't allow common user get it . For hidden agent, we allow common user get it but just filter out LLM model id ? That seems two different implementation of `hidden`. Should we use different term for this different logic for Agent, like using `is_llm_hidden_model` on Agent level, or move `is_hidden`  to Agent.llm level

Reply:
To make it simple may be we can follow the hidden model nature. As arjun said, blocking the get request for hidden agent and keep the `is_hidden` in the agent level?